parser/examples/input_parser.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use leo_errors::{emitter::Handler, Result};
+use leo_span::symbol::create_session_if_not_set_then;
+
+use std::{env, fs, path::Path};
+
+fn to_leo_tree(filepath: &Path) -> Result<String> {
+    // Loads the Leo code as a string from the given file path.
+    let program_filepath = filepath.to_path_buf();
+    let program_string = fs::read_to_string(&program_filepath).expect("failed to open input file");
+
+    // Parses the Leo file constructing an ast which is then serialized.
+    create_session_if_not_set_then(|_| {
+        let handler = Handler::default();
```

Comment:
Probably needs to do something like this: https://github.com/AleoHQ/leo/pull/1607/files#diff-1888fa155b95a6120171660a66b9fa466b1cecfe685ddba490aedbec43e41571R30